File: Backend/src/main/java/com/luckyseven/backend/domain/budget/dto/BudgetUpdateResponse.java
package com.luckyseven.backend.domain.budget.dto;

import com.luckyseven.backend.domain.budget.entity.CurrencyCode;
import jakarta.validation.constraints.NotNull;
import java.math.BigDecimal;
import java.time.LocalDateTime;
import lombok.Builder;
import lombok.Getter;

public record BudgetUpdateResponse (
    @NotNull Long id,
    @NotNull LocalDateTime updatedAt,

    @NotNull Long setBy,
    @NotNull BigDecimal balance,
    @NotNull CurrencyCode foreignCurrency,

    @NotNull BigDecimal avgExchangeRate,
    @NotNull BigDecimal foreignBalance
) { }

File: Backend/src/main/java/com/luckyseven/backend/domain/budget/dto/BudgetUpdateResponse.kt
package com.luckyseven.backend.domain.budget.dto

import com.luckyseven.backend.domain.budget.entity.CurrencyCode
import jakarta.validation.constraints.NotNull
import java.math.BigDecimal
import java.time.LocalDateTime


data class BudgetUpdateResponse(
  val id: Long,
  val updatedAt: LocalDateTime,

  val setBy: Long,
  val balance: BigDecimal,
  val foreignCurrency: CurrencyCode,

  val avgExchangeRate: BigDecimal,
  val foreignBalance: BigDecimal
)
